enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,642 @@
+  --config        The component configuration to assume is active on the cluster.
+  --version       "major.minor" version of target cluster. 
+```
+
+The "list" verb will not be provided a kubeconfig and must output tests based solely
+on environment information. If no environment arguments are provided, all tests must be listed.
+
+Listings are formatted as JSONL with one test description per line. A full listing contains
+zero or more test description objects. The following example shows an abbreviated JSONL listing 
+for three tests.
+```python
+{ "name": "T1", }
+{ "name": "T2", }
+{ "name": "T3", }
+```
+
+A test description object contains a number of attributes to inform `openshift-tests`
+discovery and planning. Note that the following annotated example would be encoded into a single 
+line of output in actual listing output.
+```python
+{
+  
+  # Base, human-readable test name. 
+  "name": "openssl version compliance",
+
+  # Optional subcomponent information.
+  "subcomponent": "ui",
+    
+  # Labels are text strings will can be used like 
+  # Ginkgo labels to group and run tests. Until labels
+  # are part of bigquery data, labels will be appended
+  # to test names with "[label]" so that name based
+  # suite construction will work. 
+  "labels": [
+    "sig-..."
+  ],
+  
+  # Tags are key=value pairs that can be used to 
+  # further classify tests.
+  "tags": {
+    "key": "value"
+  },
+  
+  "resources": {
+    # Tests can request to be run in isolation
+    # from others at different levels:
+    # "instance" - avoids being called along with a conflicting
+    #              test in a single run-test invocation.
+    # "exec"     - avoids being called at the same time in a
+    #              across all parallel extension binary invocations.
+    # "bucket"   - avoids being called in the same planning bucket.
+    # If two tests have a conflict name in common, the isolation
+    # mode will apply. "*" is a special conflict name will 
+    # ensures complete isolation in the requested mode (e.g.
+    # guaranteeing the test will be the only one to
+    # a given invocation of run-test if mode=instance).
+    "isolation": {
+      "mode": "exec",
+      "conflict":  [
+        "gpu",
+      ]
+    },
+      
+    # Tests that require significant resources can identify
+    # that fact so that the planning algorithm will not
+    # overdraw on allocatable pod resources for a set of
+    # tests run in parallel.
+    "memory": "1Gi",
+    
+    # If a duration can be roughly predicted, inclusion of this
+    # information may improve the test execution 
+    # planning algorithm performance.  
+    "duration": "2s",
+      
+    # Timeout information may also support efficient
+    # planning.
+    "timeout": "16s",
+  },
+
+  # Tests can be identified as "informing" or "production".
+  # Informing tests will not negatively impact default views
+  # in Component Readiness.

```

Comment:
I was worried about using "blocking" here because I immediately think "blocking release controller jobs." It's very possible to define a test that is mature (production), that won't run in a blocking job. I also toyed with: 
- alpha: I just want to see what this test looks like in CR.
- beta: I want this in a non-default view and raising Jiras on failure.
- production: I want this to be on the dashboard that decides releases. I could see calling this blocking because you are making a statement that broader than maturity. You are also stating how it should be used within component readiness.

Reply:
I think I still prefer the consistency of the terminology: Informing tests won't fail an individual CI job, blocking tests do.  Informing jobs don't fail a payload, blocking jobs do.

Reply:
I would lean towards informing+blocking for consistency, those terms are well established here. I can see how someone might occasionally get confused that it means the test runs in a blocking job,  but I think the alternative (correct) assumption meaning it can fail a job would be much more common.